cupy/polynomial/polynomial.py

```diff
@@ -0,0 +1,27 @@
+import cupy
+
+
+def polyvander(x, deg):
+    """Computes the Vandermonde matrix of given degree.
+
+    Args:
+        x (cupy.ndarray): array of points
+        deg (int): degree of the resulting matrix.
+
+    Returns:
+        cupy.ndarray: The Vandermonde matrix
+
+    .. seealso:: :func:`numpy.polynomial.polynomial.polyvander`
+
+    """
+    deg.__index__()
+    if deg < 0:
+        raise ValueError('deg must be non-negative')
+    if not isinstance(x, cupy.ndarray) or x.ndim != 1:
+        x = cupy.array(x, copy=False, ndmin=1)
+    x = x + 0.0
+    dims = (deg + 1,) + x.shape
+    v = cupy.ones(dims, dtype=x.dtype)
+    for i in range(1, deg + 1):
+        cupy.multiply(v[i-1], x, v[i])
```

Comment:
How about using `cupy.cumsum` to parallelize these operations?

Reply:
``` v  = cupy.cumprod(cupy.multiply(v[1:], x), axis = 0) ``` ?

Reply:
Ah, `x ** cupy.arange(deg)` seems to be better.

Reply:
I have 2 questions regarding the current implementation that I need a little help with:
1) The precision for larger size of x is different than  numpy's. Why can be the reason for this ?   
 _( I had to change some values as in `test_polyvander ` to pass the tests until finding the reason for these differences)_ 

2) The precision for complex types is always different (with the loop approach or the suggested [one](https://github.com/cupy/cupy/pull/3404#discussion_r437325708)). Why ?

( This causes tests' failure even with using `testing.numpy_cupy_allclose()` )

Reply:
The result of the computation in GPU is not guaranteed to be equal to that of CPU. For example, `powf` causes 9 ulp error in the worst case according to the [CUDA documentation](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#standard-functions).